### PR TITLE
Windows vagrant

### PR DESCRIPTION
**Pull request by cwmeijer**
Instructions on how to setup runner in VMs using Vagrant, using a Windows host. For a large part, the instructions where copied from the Ubuntu counterpart.

### CODE REVIEW

**Comment by cwmeijer** [comment]
For review, I assigned Stefan as a suggestion.

File: windows-vagrant/ansible.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[defaults]
+inventory = ./inventory.yaml
```
Comment (sverhoeven) [inline windows-vagrant/ansible.cfg:2]: In #51 we decided to call the inventory file `hosts`.
Can you rename `inventory.yaml to `hosts.example`? And add a sentence to README to copy hosts.example to hosts.

File: windows-vagrant/playbook.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      vars:
+        access_token: "{{ lookup('env','PAT') }}"
+      tags:
+        - uninstall
+    - name: Setup Docker engine
+      include_role:
+        name: nickjj.docker
+      vars:
+        docker__users:
+          - '{{ runner_user }}'
```
Comment (sverhoeven) [inline windows-vagrant/playbook.yml:24]: We noticed that installing Docker after runner can make runner error on docker usage, can you move docker task before of runner. Also we added passwordless sudo and apt update in hpc playbook can you do the same.

```suggestion
    - name: Passwordless sudo for {{ runner_user }} user 
      lineinfile:
        path: /etc/sudoers.d/{{ runner_user }}
        line: '{{ runner_user }} ALL=(ALL) NOPASSWD: ALL'
        state: present
        mode: 0440
        create: yes
        validate: 'visudo -cf %s'
    - name: Only run "update_cache=yes" if the last one is more than 3600 seconds ago
      apt:
        update_cache: yes
        cache_valid_time: 3600        
    - name: Setup Docker engine
      include_role:
        name: nickjj.docker
      vars:
        docker__users:
          - '{{ runner_user }}'
    - name: Setup GitHub actions runner
      include_role:
        name: monolithprojects.github_actions_runner
      vars:
        access_token: "{{ lookup('env','PAT') }}"
        runner_dir: /opt/actions-runner/{{ github_account }}/{{ github_repo }}
      tags:
        - uninstall
```

File: windows-vagrant/inventory.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  vars:
+    ansible_user: vagrant 
+    ansible_connection: ssh 
+    ansible_host: 127.0.0.1 
+    ansible_port: 2222 
+    ansible_ssh_private_key_file: .vagrant/machines/default/virtualbox/private_key
+    # Newer Ansible expect python to be v3, while in Ubuntu 18.04 it is still v2
+    ansible_python_interpreter: /usr/bin/python3
+    # Please update with your GitHub account/organization
+    github_account: sverhoeven
```
Comment (sverhoeven) [inline windows-vagrant/inventory.yaml:13]: Repo was moved
```suggestion
    github_account: ci-for-science
```

File: windows-vagrant/inventory.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    # Please update with your GitHub repository
+    github_repo: justatest
```
Comment (sverhoeven) [inline windows-vagrant/inventory.yaml:15]: Repo was renamed
```suggestion
    github_repo: example-python-1
```

File: windows-vagrant/README.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  IdentityFile /home/verhoes/git/NLESC-JCER/linux_actions_runner/ubuntu-vagrant/.vagrant/machines/default/virtualbox/private_key
+  IdentitiesOnly yes
+  LogLevel FATAL
+```
+
+**We will use Windows Subsystem for Linux (WSL) from from here on.**
+So to login with ssh use
+
+```shell
+ssh -i .vagrant/machines/default/virtualbox/private_key -p 2222 vagrant@127.0.0.1
```
Comment (sverhoeven) [inline windows-vagrant/README.md:58]: Add note that private_key must be copied from Windows to WSL.

**Pull request merged**